GVRf/Framework/jni/objects/animation/mesh_animation.cpp

```diff
@@ -0,0 +1,310 @@
+#include "mesh_animation.h"
+
+#include "util/gvr_log.h"
+
+#include <glm/glm.hpp>
+#include <glm/gtc/type_ptr.hpp>
+#include <glm/gtc/matrix_transform.hpp>
+
+namespace gvr {
+
+void BoneData::addBoneData(int boneId, float boneWeight) {
+
+    for (int i = 0; i < sizeof(ids); i++) {
+        if (weights[i] == 0.0) {
+            ids[i] = boneId;
+            weights[i] = boneWeight;
+            return;
+        }
+    }
+}
+
+void MeshAnimation::setScene(const aiScene* scene) {
+    aiScene_ = scene;
+}
+
+void MeshAnimation::setGlobalInverTransform(const glm::mat4& globalTransform) {
```

Comment:
This is from fredduarte's code. Because we use full names, I will add a patch to fix it.